lib/arclight/custom_component.rb

```diff
@@ -28,11 +28,20 @@ def to_solr(solr_doc = {})
       Solrizer.insert_field(solr_doc, 'level', formatted_level, :facetable) # human-readable for facet `level_sim`
       Solrizer.insert_field(solr_doc, 'date_range', formatted_unitdate_for_range, :facetable)
       Solrizer.insert_field(solr_doc, 'access_subjects', access_subjects, :facetable)
+      resolve_repository(solr_doc)
       solr_doc
     end
 
     private
 
+    attr_accessor :repository
+
+    def resolve_repository(solr_doc)
+      @repository ||= solr_doc['repository_ssm']
```

Comment:
Yeah, I struggled with this piece of the code. I put `repository_as_configured` as a shared method with no parameters which yielded this requirement for having a repository accessor. The document already has one (defined by the terminology) but the component does not. So that's why it's like that on lines 37-40.

I can parameterize that shared function if it helps.